metadata-io/src/main/java/com/linkedin/metadata/dataproducts/sideeffects/DataProductUnsetSideEffect.java

```diff
@@ -93,40 +95,45 @@ private static Stream<MCPItem> generatePatchRemove(
         if (!result.getEntities().isEmpty()) {
           for (RelatedEntities entity : result.getEntities()) {
             if (!mclItem.getUrn().equals(UrnUtils.getUrn(entity.getSourceUrn()))) {
-              EntitySpec entitySpec =
-                  retrieverContext
-                      .getAspectRetriever()
-                      .getEntityRegistry()
-                      .getEntitySpec(DATA_PRODUCT_ENTITY_NAME);
               GenericJsonPatch.PatchOp patchOp = new GenericJsonPatch.PatchOp();
               patchOp.setOp(PatchOperationType.REMOVE.getValue());
               patchOp.setPath(String.format("/assets/%s", entity.getDestinationUrn()));
-              mcpItems.add(
-                  PatchItemImpl.builder()
-                      .urn(UrnUtils.getUrn(entity.getSourceUrn()))
-                      .entitySpec(
-                          retrieverContext
-                              .getAspectRetriever()
-                              .getEntityRegistry()
-                              .getEntitySpec(DATA_PRODUCT_ENTITY_NAME))
-                      .aspectName(DATA_PRODUCT_PROPERTIES_ASPECT_NAME)
-                      .aspectSpec(entitySpec.getAspectSpec(DATA_PRODUCT_PROPERTIES_ASPECT_NAME))
-                      .patch(
-                          GenericJsonPatch.builder()
-                              .arrayPrimaryKeys(
-                                  Map.of(
-                                      DataProductPropertiesTemplate.ASSETS_FIELD_NAME,
-                                      List.of(DataProductPropertiesTemplate.KEY_FIELD_NAME)))
-                              .patch(List.of(patchOp))
-                              .build()
-                              .getJsonPatch())
-                      .auditStamp(mclItem.getAuditStamp())
-                      .systemMetadata(mclItem.getSystemMetadata())
-                      .build(retrieverContext.getAspectRetriever().getEntityRegistry()));
+              patchOpMap
+                  .computeIfAbsent(entity.getSourceUrn(), urn -> new ArrayList<>())
+                  .add(patchOp);
             }
           }
         }
       }
+      for (String urn : patchOpMap.keySet()) {
```

Comment:
We probably want to batch this in some way. I know that today, data products are constrained to a single aspect, however if there might be 1 million assets, it could generate a patch that is too large.